framework/src/postprocessors/Postprocessor.C

```diff
@@ -64,3 +65,101 @@ Postprocessor::getValue() const
 {
   mooseError("getValue() (const or non-const) must be implemented.");
 }
+
+typename Postprocessor::ValueType
+Postprocessor::evaluate(const ElemArg & /*elem_arg*/, const Moose::StateArg & /*state*/) const
+{
+  return getValue();
+}
+
+typename Postprocessor::ValueType
+Postprocessor::evaluate(const FaceArg & /*face*/, const Moose::StateArg & /*state*/) const
+{
+  return getValue();
+}
+
+typename Postprocessor::ValueType
+Postprocessor::evaluate(const ElemQpArg & /*elem_qp*/, const Moose::StateArg & /*state*/) const
+{
+  return getValue();
+}
+
+typename Postprocessor::ValueType
+Postprocessor::evaluate(const ElemSideQpArg & /*elem_side_qp*/,
+                        const Moose::StateArg & /*state*/) const
+{
+  return getValue();
+}
+
+typename Postprocessor::ValueType
+Postprocessor::evaluate(const ElemPointArg & /*elem_point_arg*/,
+                        const Moose::StateArg & /*state*/) const
+{
+  return getValue();
+}
+
+typename Postprocessor::GradientType
+Postprocessor::evaluateGradient(const ElemArg & /*elem_arg*/,
+                                const Moose::StateArg & /*state*/) const
+{
+  return 0;
+}
+
+typename Postprocessor::GradientType
+Postprocessor::evaluateGradient(const FaceArg & /*face*/, const Moose::StateArg & /*state*/) const
+{
+  return 0;
+}
+
+typename Postprocessor::GradientType
+Postprocessor::evaluateGradient(const ElemQpArg & /*elem_qp*/,
+                                const Moose::StateArg & /*state*/) const
+{
+  return 0;
+}
+
+typename Postprocessor::GradientType
+Postprocessor::evaluateGradient(const ElemSideQpArg & /*elem_side_qp*/,
+                                const Moose::StateArg & /*state*/) const
+{
+  return 0;
+}
+
+typename Postprocessor::GradientType
+Postprocessor::evaluateGradient(const ElemPointArg & /*elem_point_arg*/,
+                                const Moose::StateArg & /*state*/) const
+{
+  return 0;
+}
+
+typename Postprocessor::DotType
+Postprocessor::evaluateDot(const ElemArg & /*elem_arg*/, const Moose::StateArg & /*state*/) const
+{
+  return 0;
```

Comment:
I d rather an error for time derivatives
or a doOnce warning at least

Reply:
Yeah, a doOnce sounds like a good idea. I think I'll do the same for the gradient.

Reply:
no not gradient imo
a PP is assumed to have a 0 gradient, that's ok